src/Kanbas/Courses/Quizzes/Preview/index.tsx

```diff
@@ -5,26 +5,94 @@ import QuizQuestionPreviewSingle from "./Question/Single";
 import QuizQuestionPreviewList from "./Question/List";
 import MiddleContentActions from "../../../layout/Content/NotLeftSide/MiddleContent/MiddleContentActions";
 import MiddleContentData from "../../../layout/Content/NotLeftSide/MiddleContent/MiddleContentData";
+import { IoAlertCircleOutline } from "react-icons/io5";
+import { useDispatch, useSelector } from "react-redux";
+import { KanbasState } from "../../../store";
+import { Button, Form } from "react-bootstrap";
+import "./index.css";
+import { useEffect, useRef, useState } from "react";
+import { IKanbasQuizQuestion } from "../../../store/interfaces/quizzes";
+import { useParams } from "react-router";
+import { getQuizQuestions } from "../client";
+import { resetPreview, setQuizQuestions } from "../reducer";
+import { getCurrentHumanReadableDate } from "../../common/Utils";
 
 function QuizPreview() {
 
-    // TODO this variable will be removed later to use props directly from JSON
-    const isQneQuestionAtATime = false;
+    const { quizId } = useParams();
+    const dispatch = useDispatch();
+    const questionRefs = useRef<(HTMLElement | null)[]>([]);
+
+    const [lastSavedTime, setLastSavedTime] = useState(new Date());
+
+    const quizDetails = useSelector(
+        (state: KanbasState) => state.quizzesReducer.quiz
+    );
+
+    const scrollToQuestion = (index: number) => {
+        const questionRef = questionRefs.current[index];
+        questionRef?.scrollIntoView({ behavior: 'smooth' });
+    }
+
+    const handleSubmit = () => {
+    }
+
+    useEffect(() => {
+        if (quizId !== undefined) {
+            getQuizQuestions(quizId).then((questions: IKanbasQuizQuestion[]) => {
+                dispatch(setQuizQuestions(questions));
+                dispatch(resetPreview());
+            });
+        };
+    }, [dispatch]);
+
+    useEffect(() => {
+        const id = setInterval(() => {
+            setLastSavedTime(new Date());
+        }, 1000); // update every second
```

Comment:
We shouldn't need a timer for this. This should update every time a action goes through, like next, prev, etc or a state change occurs.